modules/programs/zsh.nix

```diff
@@ -79,6 +100,18 @@ in
 
       home.file.".zshrc".text = ''
         ${export "HISTSIZE" cfg.historySize}
+        ${export "HISTFILE" cfg.historyFile}
+
+        setopt HIST_FCNTL_LOCK
+        ${if cfg.historyIgnoreDups then "setopt" else "unsetopt"} HIST_IGNORE_DUPS
+        ${if cfg.historyShare then "setopt" else "unsetopt"} SHARE_HISTORY
+
+        # Tell zsh how to find installed completions
+        for p in ''${(z)NIX_PROFILES}; do
+          fpath+=($p/share/zsh/site-functions $p/share/zsh/$ZSH_VERSION/functions)
+        done
+
+        HELPDIR="${pkgs.zsh}/share/zsh/$ZSH_VERSION/help"
```

Comment:
Currently, it is fixed in [zsh module](https://github.com/NixOS/nixpkgs/blob/release-17.03/nixos/modules/programs/zsh/zsh.nix#L134) and not in the package if I understand correctly. But yes, it looks like something that should work with installed package and should not require module activation. Should I remove it from this PR? Not sure about `fpath` though.